### PR TITLE
Access rights bug in usermanagemant fixed

### LINKED ISSUE #1411: Access rights
Only the super user should be able to add users in all tenants, and the tenant admin  should be able to see only the users and tenants to which he/she is an admin. The other users should not be able to view the user management link and should get access denied message.

### PR DESCRIPTION
**Pull request by taylortom**
Fixes #1411 

@aniketdharia, @ambadasavhad have cherry picked what I believe is the fix, can you confirm?

### CODE REVIEW

File: frontend/src/plugins/userManagement/views/userManagementView.js [hunk 1]
```diff
@@ -64,20 +64,48 @@ define(function(require){
     },
 
     createUserView: function(model) {
+      if (!this.isCurrentUserSuperadmin) {
+        if (model.get('_tenantId')._id !== this.currentUser.get('_tenantId')._id) {
```
Comment (taylortom) [inline frontend/src/plugins/userManagement/views/userManagementView.js]: This breaks if you have a user with no tenant.

File: frontend/src/plugins/userManagement/views/userManagementView.js [hunk 2]
```diff
@@ -64,20 +64,48 @@ define(function(require){
...
-  }, {
-    template: 'userManagement'
-  });
+    setCurrentUsersRole:function(){
+      var self = this;
+      var currentUserId = Origin.sessionModel.get('id');
+      this.isCurrentUserTenantadmin = false;
+      this.isCurrentUserSuperadmin = false;
+
+      this.currentUser = _.find(this.users.models, function(user) {
```
Comment (aniketdharia) [inline frontend/src/plugins/userManagement/views/userManagementView.js]: @ambadasavhad please fix this

File: frontend/src/plugins/userManagement/views/userManagementView.js (hunk 1 above)
Comment (aniketdharia) [inline frontend/src/plugins/userManagement/views/userManagementView.js]: @ambadasavhad please fix this

File: frontend/src/plugins/userManagement/views/addUserView.js [hunk 3]
```diff
@@ -3,15 +3,32 @@ define(function(require){
...
+  var UserCollection = require('../collections/userCollection.js');
 
   var AddUserView = OriginView.extend({
     tagName: 'div',
     className: 'addUser',
     createdUserId: false,
+    settings: {
+      autoRender: false
+    },
+    users: new UserCollection(),
```
Comment (aniketdharia) [inline frontend/src/plugins/userManagement/views/addUserView.js]: @ambadasavhad please fix this

File: frontend/src/plugins/userManagement/views/addUserView.js [hunk 4]
```diff
@@ -91,10 +108,34 @@ define(function(require){
         title: "Couldn't add user",
         text: data.responseText || error
       });
+    },
+
+    isCurrentUserTenantAdmin:function(){
```
Comment (aniketdharia) [inline frontend/src/plugins/userManagement/views/addUserView.js]: @ambadasavhad please fix this

**Comment by canstudios-nicolaw** [comment]
Hi @ambadasavhad @aniketdharia Please could you tell me what the current status of this work is?

**Comment by aniketdharia** [comment]
@canstudios-nicolaw it is done.. we will commit b4 this weekend.

**Comment by canstudios-nicolaw** [comment]
@aniketdharia Brilliant, thanks!

**Comment by ambadasavhad** [comment]
already fixed in #1411

**Comment by taylortom** [comment]
Closing this, as no longer relevant

**Pull request closed without merging**